(base) C:\repos\Listen-Attend-Spell-v2>python -m torch.distributed.launch train.py
NOTE: Redirects are currently not supported in Windows or MacOs.
C:\ProgramData\Miniconda3\lib\site-packages\torch\distributed\launch.py:163: DeprecationWarning: The 'warn' method is deprecated, use 'warning' instead
  logger.warn(
The module torch.distributed.launch is deprecated and going to be removed in future.Migrate to torch.distributed.run
INFO:torch.distributed.run:Using nproc_per_node=auto, seting to 1 since the instance has 4 gpu
WARNING:torch.distributed.run:--use_env is deprecated and will be removed in future releases.
 Please read local_rank from `os.environ('LOCAL_RANK')` instead.
INFO:torch.distributed.launcher.api:Starting elastic_operator with launch configs:
  entrypoint       : train.py
  min_nodes        : 1
  max_nodes        : 1
  nproc_per_node   : 1
  run_id           : none
  rdzv_backend     : static
  rdzv_endpoint    : 127.0.0.1:29500
  rdzv_configs     : {'rank': 0, 'timeout': 900}
  max_restarts     : 3
  monitor_interval : 5
  log_dir          : None
  metrics_cfg      : {}

INFO:torch.distributed.elastic.agent.server.local_elastic_agent:log directory set to: C:\Users\TPOEK\AppData\Local\Temp\torchelastic_8cqygd75\none_da50ehei
INFO:torch.distributed.elastic.agent.server.api:[default] starting workers for entrypoint: python.exe
INFO:torch.distributed.elastic.agent.server.api:[default] Rendezvous'ing worker group
C:\ProgramData\Miniconda3\lib\site-packages\torch\distributed\elastic\utils\store.py:52: FutureWarning: This is an experimental API and will be changed in future.
  warnings.warn(
INFO:torch.distributed.elastic.agent.server.api:[default] Rendezvous complete for workers. Result:
  restart_count=0
  master_addr=127.0.0.1
  master_port=29500
  group_rank=0
  group_world_size=1
  local_ranks=[0]
  role_ranks=[0]
  global_ranks=[0]
  role_world_sizes=[1]
  global_world_sizes=[1]

INFO:torch.distributed.elastic.agent.server.api:[default] Starting worker group
INFO:torch.distributed.elastic.multiprocessing:Setting worker0 reply file to: C:\Users\TPOEK\AppData\Local\Temp\torchelastic_8cqygd75\none_da50ehei\attempt_0\0\error.json
usage: train.py [-h] [--input-dim INPUT_DIM]
                [--encoder-hidden-size ENCODER_HIDDEN_SIZE]
                [--decoder-hidden-size DECODER_HIDDEN_SIZE]
                [--num-layers NUM_LAYERS] [--embedding-dim EMBEDDING_DIM]
                [--end-epoch END_EPOCH] [--lr LR] [--lr-step LR_STEP]
                [--optimizer OPTIMIZER] [--weight-decay WEIGHT_DECAY]
                [--mom MOM] [--emb-size EMB_SIZE] [--batch-size BATCH_SIZE]
                [--checkpoint CHECKPOINT]
train.py: error: unrecognized arguments: --local_rank=0
ERROR:torch.distributed.elastic.multiprocessing.api:failed (exitcode: 2) local_rank: 0 (pid: 17868) of binary: C:\ProgramData\Miniconda3\python.exe
ERROR:torch.distributed.elastic.agent.server.local_elastic_agent:[default] Worker group failed
INFO:torch.distributed.elastic.agent.server.api:[default] Worker group FAILED. 3/3 attempts left; will restart worker group
INFO:torch.distributed.elastic.agent.server.api:[default] Stopping worker group
INFO:torch.distributed.elastic.agent.server.api:[default] Rendezvous'ing worker group
INFO:torch.distributed.elastic.agent.server.api:[default] Rendezvous complete for workers. Result:
  restart_count=1
  master_addr=127.0.0.1
  master_port=29500
  group_rank=0
  group_world_size=1
  local_ranks=[0]
  role_ranks=[0]
  global_ranks=[0]
  role_world_sizes=[1]
  global_world_sizes=[1]

INFO:torch.distributed.elastic.agent.server.api:[default] Starting worker group
INFO:torch.distributed.elastic.multiprocessing:Setting worker0 reply file to: C:\Users\TPOEK\AppData\Local\Temp\torchelastic_8cqygd75\none_da50ehei\attempt_1\0\error.json
usage: train.py [-h] [--input-dim INPUT_DIM]
                [--encoder-hidden-size ENCODER_HIDDEN_SIZE]
                [--decoder-hidden-size DECODER_HIDDEN_SIZE]
                [--num-layers NUM_LAYERS] [--embedding-dim EMBEDDING_DIM]
                [--end-epoch END_EPOCH] [--lr LR] [--lr-step LR_STEP]
                [--optimizer OPTIMIZER] [--weight-decay WEIGHT_DECAY]
                [--mom MOM] [--emb-size EMB_SIZE] [--batch-size BATCH_SIZE]
                [--checkpoint CHECKPOINT]
train.py: error: unrecognized arguments: --local_rank=0
ERROR:torch.distributed.elastic.multiprocessing.api:failed (exitcode: 2) local_rank: 0 (pid: 9992) of binary: C:\ProgramData\Miniconda3\python.exe
ERROR:torch.distributed.elastic.agent.server.local_elastic_agent:[default] Worker group failed
INFO:torch.distributed.elastic.agent.server.api:[default] Worker group FAILED. 2/3 attempts left; will restart worker group
INFO:torch.distributed.elastic.agent.server.api:[default] Stopping worker group
INFO:torch.distributed.elastic.agent.server.api:[default] Rendezvous'ing worker group
INFO:torch.distributed.elastic.agent.server.api:[default] Rendezvous complete for workers. Result:
  restart_count=2
  master_addr=127.0.0.1
  master_port=29500
  group_rank=0
  group_world_size=1
  local_ranks=[0]
  role_ranks=[0]
  global_ranks=[0]
  role_world_sizes=[1]
  global_world_sizes=[1]

INFO:torch.distributed.elastic.agent.server.api:[default] Starting worker group
INFO:torch.distributed.elastic.multiprocessing:Setting worker0 reply file to: C:\Users\TPOEK\AppData\Local\Temp\torchelastic_8cqygd75\none_da50ehei\attempt_2\0\error.json
usage: train.py [-h] [--input-dim INPUT_DIM]
                [--encoder-hidden-size ENCODER_HIDDEN_SIZE]
                [--decoder-hidden-size DECODER_HIDDEN_SIZE]
                [--num-layers NUM_LAYERS] [--embedding-dim EMBEDDING_DIM]
                [--end-epoch END_EPOCH] [--lr LR] [--lr-step LR_STEP]
                [--optimizer OPTIMIZER] [--weight-decay WEIGHT_DECAY]
                [--mom MOM] [--emb-size EMB_SIZE] [--batch-size BATCH_SIZE]
                [--checkpoint CHECKPOINT]
train.py: error: unrecognized arguments: --local_rank=0
ERROR:torch.distributed.elastic.multiprocessing.api:failed (exitcode: 2) local_rank: 0 (pid: 12528) of binary: C:\ProgramData\Miniconda3\python.exe
ERROR:torch.distributed.elastic.agent.server.local_elastic_agent:[default] Worker group failed
INFO:torch.distributed.elastic.agent.server.api:[default] Worker group FAILED. 1/3 attempts left; will restart worker group
INFO:torch.distributed.elastic.agent.server.api:[default] Stopping worker group
INFO:torch.distributed.elastic.agent.server.api:[default] Rendezvous'ing worker group
INFO:torch.distributed.elastic.agent.server.api:[default] Rendezvous complete for workers. Result:
  restart_count=3
  master_addr=127.0.0.1
  master_port=29500
  group_rank=0
  group_world_size=1
  local_ranks=[0]
  role_ranks=[0]
  global_ranks=[0]
  role_world_sizes=[1]
  global_world_sizes=[1]

INFO:torch.distributed.elastic.agent.server.api:[default] Starting worker group
INFO:torch.distributed.elastic.multiprocessing:Setting worker0 reply file to: C:\Users\TPOEK\AppData\Local\Temp\torchelastic_8cqygd75\none_da50ehei\attempt_3\0\error.json
usage: train.py [-h] [--input-dim INPUT_DIM]
                [--encoder-hidden-size ENCODER_HIDDEN_SIZE]
                [--decoder-hidden-size DECODER_HIDDEN_SIZE]
                [--num-layers NUM_LAYERS] [--embedding-dim EMBEDDING_DIM]
                [--end-epoch END_EPOCH] [--lr LR] [--lr-step LR_STEP]
                [--optimizer OPTIMIZER] [--weight-decay WEIGHT_DECAY]
                [--mom MOM] [--emb-size EMB_SIZE] [--batch-size BATCH_SIZE]
                [--checkpoint CHECKPOINT]
train.py: error: unrecognized arguments: --local_rank=0
ERROR:torch.distributed.elastic.multiprocessing.api:failed (exitcode: 2) local_rank: 0 (pid: 6324) of binary: C:\ProgramData\Miniconda3\python.exe
ERROR:torch.distributed.elastic.agent.server.local_elastic_agent:[default] Worker group failed
INFO:torch.distributed.elastic.agent.server.api:Local worker group finished (FAILED). Waiting 300 seconds for other agents to finish
C:\ProgramData\Miniconda3\lib\site-packages\torch\distributed\elastic\utils\store.py:70: FutureWarning: This is an experimental API and will be changed in future.
  warnings.warn(
INFO:torch.distributed.elastic.agent.server.api:Done waiting for other agents. Elapsed: 0.0 seconds
{"name": "torchelastic.worker.status.FAILED", "source": "WORKER", "timestamp": 0, "metadata": {"run_id": "none", "global_rank": 0, "group_rank": 0, "worker_id": "6324", "role": "default", "hostname": "WGC1NJ9LLJ5T2.nanjingrec.ford.com", "state": "FAILED", "total_run_time": 20, "rdzv_backend": "static", "raw_error": "{\"message\": \"<NONE>\"}", "metadata": "{\"group_world_size\": 1, \"entry_point\": \"python.exe\", \"local_rank\": [0], \"role_rank\": [0], \"role_world_size\": [1]}", "agent_restarts": 3}}
{"name": "torchelastic.worker.status.SUCCEEDED", "source": "AGENT", "timestamp": 0, "metadata": {"run_id": "none", "global_rank": null, "group_rank": 0, "worker_id": null, "role": "default", "hostname": "WGC1NJ9LLJ5T2.nanjingrec.ford.com", "state": "SUCCEEDED", "total_run_time": 20, "rdzv_backend": "static", "raw_error": null, "metadata": "{\"group_world_size\": 1, \"entry_point\": \"python.exe\"}", "agent_restarts": 3}}
C:\ProgramData\Miniconda3\lib\site-packages\torch\distributed\elastic\multiprocessing\errors\__init__.py:354: UserWarning:

**********************************************************************
               CHILD PROCESS FAILED WITH NO ERROR_FILE
**********************************************************************
CHILD PROCESS FAILED WITH NO ERROR_FILE
Child process 6324 (local_rank 0) FAILED (exitcode 2)
Error msg: Process failed with exitcode 2
Without writing an error file to <N/A>.
While this DOES NOT affect the correctness of your application,
no trace information about the error will be available for inspection.
Consider decorating your top level entrypoint function with
torch.distributed.elastic.multiprocessing.errors.record. Example:

  from torch.distributed.elastic.multiprocessing.errors import record

  @record
  def trainer_main(args):
     # do train
**********************************************************************
  warnings.warn(_no_error_file_warning_msg(rank, failure))
Traceback (most recent call last):
  File "C:\ProgramData\Miniconda3\lib\runpy.py", line 194, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "C:\ProgramData\Miniconda3\lib\runpy.py", line 87, in _run_code
    exec(code, run_globals)
  File "C:\ProgramData\Miniconda3\lib\site-packages\torch\distributed\launch.py", line 173, in <module>
    main()
  File "C:\ProgramData\Miniconda3\lib\site-packages\torch\distributed\launch.py", line 169, in main
    run(args)
  File "C:\ProgramData\Miniconda3\lib\site-packages\torch\distributed\run.py", line 621, in run
    elastic_launch(
  File "C:\ProgramData\Miniconda3\lib\site-packages\torch\distributed\launcher\api.py", line 116, in __call__
    return launch_agent(self._config, self._entrypoint, list(args))
  File "C:\ProgramData\Miniconda3\lib\site-packages\torch\distributed\elastic\multiprocessing\errors\__init__.py", line 348, in wrapper
    return f(*args, **kwargs)
  File "C:\ProgramData\Miniconda3\lib\site-packages\torch\distributed\launcher\api.py", line 245, in launch_agent
    raise ChildFailedError(
torch.distributed.elastic.multiprocessing.errors.ChildFailedError:
***************************************
            train.py FAILED
=======================================
Root Cause:
[0]:
  time: 2021-08-18_09:35:13
  rank: 0 (local_rank: 0)
  exitcode: 2 (pid: 6324)
  error_file: <N/A>
  msg: "Process failed with exitcode 2"
=======================================
Other Failures:
  <NO_OTHER_FAILURES>
***************************************

